moebot_bot/bot/commands/channelRotationScheduler.go

```diff
@@ -0,0 +1,206 @@
+package commands
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/bwmarrin/discordgo"
+	"github.com/camd67/moebot/moebot_bot/util"
+	"github.com/camd67/moebot/moebot_bot/util/db"
+	"github.com/camd67/moebot/moebot_bot/util/moeDiscord"
+)
+
+type ChannelRotationScheduler struct {
+	schedulerType db.SchedulerType
+	session       *discordgo.Session
+}
+
+func NewChannelRotationScheduler(schedulerType db.SchedulerType, session *discordgo.Session) *ChannelRotationScheduler {
+	return &ChannelRotationScheduler{schedulerType, session}
+}
+
+func (s *ChannelRotationScheduler) Execute(operationId int64) {
+	channelRotation, err := db.ChannelRotationQuery(operationId)
+	if err != nil {
+		log.Println("Failed to retrieve operation informations (operation is possibly being created) ", err)
+		return
+	}
+	server, err := db.ServerQueryById(channelRotation.ServerID)
+	if err != nil {
+		log.Println("Failed to retrieve server informations ", err)
```

Comment:
For most of the errors in here, please include relevant information, such as `ServerID` in this case so it's easier to track down later. Same with `GuildUID` in the next error and so on